### PR TITLE
Remove Client Validation

### PR DESCRIPTION
**Pull request by SasinduDilshara**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by SasinduDilshara** [comment]
With this change
https://github.com/ballerina-platform/ballerina-lang/blob/3d23ee1d95b4f76be6aa83a9abdfd5575e508610/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java#L4958
and
https://github.com/ballerina-platform/ballerina-lang/blob/3d23ee1d95b4f76be6aa83a9abdfd5575e508610/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java#L4970
checks will passed for `Client`

File: compiler/ballerina-lang/src/main/resources/compiler.properties [hunk 1]
```diff
@@ -1946,7 +1946,7 @@ error.unsupported.exposure.of.construct.from.module.generated.for.client.decl=\
   exposing a construct from a module generated for a client declaration is not yet supported
 
 error.module.generated.for.client.decl.must.have.a.client.object.type=\
-  a module generated for a client declaration must have an object type or class named ''client''
+  a module generated for a client declaration must have an object type or class named ''Client''
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/resources/compiler.properties:1949]: You will also have to change this to get the tests passing: https://github.com/ballerina-platform/ballerina-lang/commit/9757d0847d626bb3d98d6e25a062e4875c13d0de

Comment (SasinduDilshara) [reply compiler/ballerina-lang/src/main/resources/compiler.properties:1949]: Updated.

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/plugins/IDLClientCompilerTests.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ public class IDLClientCompilerTests {
     private static final String UNSUPPORTED_EXPOSURE_OF_PUBLIC_CONSTRUCT_ERROR =
             "exposing a construct from a module generated for a client declaration is not yet supported";
     private static final String NO_CLIENT_OBJECT_NAMED_CLIENT_IN_GENERATED_MODULE_ERROR =
-            "a module generated for a client declaration must have an object type or class named 'client'";
+            "a module generated for a client declaration must have an object type or class named 'Client'";
```
Comment (azinneera) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/plugins/IDLClientCompilerTests.java:65]: We need to change the generated client used for this test class: https://github.com/ballerina-platform/ballerina-lang/blob/d956272741ca0b2f5b765a729e8fcf0f0304f047/project-api/test-artifacts/idl-import-simple-compiler-plugin/src/main/java/io/idl/plugins/simpleclient/SimpleClientGeneratorPlugin.java#L101

Comment (SasinduDilshara) [reply project-api/project-api-test/src/test/java/io/ballerina/projects/test/plugins/IDLClientCompilerTests.java:65]: will change

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java [hunk 3]
```diff
@@ -4946,7 +4945,6 @@ private void validateClientDeclarations(BLangPackage bLangPackage, AnalyzerData
             for (Map.Entry<String, Map<LineRange, Optional<PackageID>>> compUnitEntry :
                     pkgIdEntry.getValue().entrySet()) {
                 if (compUnitEntry.getValue().containsValue(Optional.of(bLangPackage.packageID))) {
-                    checkForClientObjectTypeOrClass(bLangPackage);
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java:4949]: Are we removing the check for a client object altogether? Don't we still need to check if there's at least a client object, maybe with the name `Client`?

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/HierarchicalPackageNameTests.java [hunk 4]
```diff
@@ -134,7 +134,7 @@ public void testResolveHierarchicalPackageInDist() {
             if (importModuleResponse.resolutionStatus().equals(ResolutionResponse.ResolutionStatus.UNRESOLVED)) {
                 Assert.assertEquals(importModuleResponse.importModuleRequest(), request4);
             } else if (importModuleResponse.importModuleRequest().moduleName().equals("a.c")) {
-                Assert.assertEquals(importModuleResponse.packageDescriptor().name().toString(), "a");
+                Assert.assertEquals(importModuleResponse.packageDescriptor().name().toString(), "a.c");
```
Comment (MaryamZi) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/HierarchicalPackageNameTests.java:137]: Are these unrelated changes to tests required?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java (hunk 3 above)
Comment (hasithaa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java:4949]: There is a miscommunication, We need this validation. But we need to remove where we give this error "invalid usage of the 'client' keyword as an unquoted identifier in a qualified identifier: allowed only with client declarations"

Comment (MaryamZi) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java:4949]: Isn't the requirement disallowing `foo:client` altogether for now?

If so, IIRC, we'll just have to remove the client keyword from `parseQualifiedIdentifier` (https://github.com/ballerina-platform/ballerina-lang/pull/37471/files#diff-c6651eecefb7bdd9c3fbaea34bbad99af723566103b5a47b66da71a412918c5cR4185).

That should result in a parser error, and then we can remove the invalid usage error too (to avoid an additional error for `foo:client` when `foo` isn't defined with a client declaration).

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/plugins/IDLClientCompilerTests.java (hunk 2 above)
Comment (SasinduDilshara) [reply project-api/project-api-test/src/test/java/io/ballerina/projects/test/plugins/IDLClientCompilerTests.java]: Updated

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java (hunk 3 above)
Comment (SasinduDilshara) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java]: Updated

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/HierarchicalPackageNameTests.java (hunk 4 above)
Comment (SasinduDilshara) [reply project-api/project-api-test/src/test/java/io/ballerina/projects/test/HierarchicalPackageNameTests.java]: Updated

**Review by gimantha — APPROVED** [review]
LGTM

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/CreateVariableTest.java [hunk 5]
```diff
@@ -96,7 +96,7 @@ public Object[][] dataProvider() {
                 {"variableAssignmentRequiredCodeAction43.json"},
                 {"variableAssignmentRequiredCodeAction44.json"},
                 {"variableAssignmentRequiredCodeAction45.json"},
-                {"variableAssignmentRequiredCodeAction46.json"},
+                // {"variableAssignmentRequiredCodeAction46.json"},
```
Comment (gimantha) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/CreateVariableTest.java:99]: Will this be fixed from LS side?

**Review by hasithaa — APPROVED** [review]
Looks good to me

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/CreateVariableTest.java (hunk 5 above)
Comment (SasinduDilshara) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/CreateVariableTest.java:99]: Yes they add a separat PR
